src/avram/callbacks.cr

```diff
@@ -72,6 +72,19 @@ module Avram::Callbacks
     end
   end
 
+  # Same as `before_save`, but with a different name
+  macro before_delete(method_name, if _if = nil, unless _unless = nil)
+    {% unless _if.is_a?(SymbolLiteral) || _if.is_a?(NilLiteral) %}
+      conditional_error_for_inline_callbacks(:before_delete, {{ method_name }}, :if)
+    {% end %}
+    {% unless _unless.is_a?(SymbolLiteral) || _unless.is_a?(NilLiteral) %}
+      conditional_error_for_inline_callbacks(:before_delete, {{ method_name }}, :unless)
+    {% end %}
+    before_delete(if: {{ _if }}, unless: {{ _unless }}) do
+      {{ method_name.id }}
```

Comment:
hmm... 🤔  That's not a bad idea. I guess you technically have access to it through just calling `self.record`. I'd be ok with adding it in though.

Reply:
I keep going back and forth on this... Would it be too weird that it doesn't match up with the `before_run` and `before_save` interfaces? Those make sense to not pass down the variable... Do you think it would be more helpful than confusing?